app/lib/task/backend.dart

```diff
@@ -900,6 +905,45 @@ class TaskBackend {
     }
   }
 
+  /// Get status information for a package being analyzed.
+  Future<PackageStatus> packageStatus(String package) async {
+    final status = await cache.taskPackageStatus(package).get(() async {
+      final key = PackageState.createKey(_db, runtimeVersion, package);
+      final state = await dbService.lookupOrNull<PackageState>(key);
+
+      return PackageStatus(
+        package: package,
+        versions: (state?.versions ?? {}).entries.map((e) {
+          final v = e.key;
+          final s = e.value;
+
+          var status = TaskPackageVersionStatus.completed;
+          if (s.attempts > 0 && s.attempts < taskRetryLimit) {
+            // attempts > 0 && attempts < taskRetryLimit  ==> pending
+            status = TaskPackageVersionStatus.pending;
+          } else if (s.scheduled
+              .add(Duration(days: 31))
+              .isBefore(clock.now())) {
+            // scheduled + 31 days < now                  ==> pending
+            status = TaskPackageVersionStatus.pending;
+          } else if (s.attempts == 0 && s.scheduled == initialTimestamp) {
```

Comment:
Maybe move this pending to the top of the if conditional? It seems like this is the default state.